contracts/InterchainTokenService.sol

```diff
@@ -901,20 +869,21 @@ contract InterchainTokenService is
         bytes32 commandId,
         string calldata sourceChain,
         string calldata sourceAddress,
-        bytes memory payload,
-        bytes32 payloadHash
-    ) internal {
+        bytes calldata payload
+    ) internal override whenNotPaused {
+        bytes32 payloadHash = keccak256(payload);
         uint256 messageType;
         string memory originalSourceChain;
-        (messageType, originalSourceChain, payload) = _getExecuteParams(sourceChain, payload);
+        bytes memory originalPayload;
```

Comment:
this is not storing the original payload. this is the transformed payload. although, reverting the executable inheritance should revert the change here anyways

Reply:
also, what is the point of introducing the extra local variable? The payload that is preserved does not seem to be used anywhere.